submissions/api.py

```diff
@@ -684,6 +686,23 @@ def reset_score(student_id, course_id, item_id):
             item_id=item_id,
         )
 
+        if clear_state:
+            for sub in student_item.submission_set.all():
+                # soft-delete the Submission
+                sub.status = Submission.DELETED
+                sub.save(update_fields=["status"])
+
+                # Also clear out cached values
+                cache_key = "submissions.submission.{}".format(sub.uuid)
```

Comment:
I'd have a function like `get_submission_cache_key` (or `make...` or `create...`), following whatever naming convention we've established for something like this.  This function should be used all places the cache key is needed.